test/gtest/ucp/test_ucp_am.cc

```diff
@@ -1902,20 +1909,72 @@ class test_ucp_am_nbx_rndv_memtype : public test_ucp_am_nbx_rndv {
     }
 
 private:
+    static void base_test_generator(variant_vec_t &variants)
+    {
+        // Do not instantiate test case if no GPU memtypes supported.
+        if (!mem_buffer::is_gpu_supported()) {
+            return;
+        }
+
+        add_variant_memtypes(variants,
+                             test_ucp_am_nbx_prereg::get_test_variants);
+    }
+
+    static void add_variant_memtypes(variant_vec_t &variants,
+                                     get_variants_func_t generator)
+    {
+        ucp_test::add_variant_memtypes(variants, generator);
```

Comment:
where do we add HOST memory?

Reply:
`add_variant_memtypes` iterates through the vector returned by `supported_mem_types`. And host memory type is added as a first element to the vector in `supported_mem_types`.